assets/app/views/edit/build-config.html

```diff
@@ -0,0 +1,475 @@
+                            rendererOptions: {
+                              fadeFoldWidgets: true,
+                              showPrintMargin: false
+                            }
+                          }" ng-model="updatedBuildConfig.spec.source.dockerfile" class="dockerfile-editor dockerfile-mode"></div>
+                        </div>
+
+                        <div class="form-group" ng-if="!!updatedBuildConfig.spec.strategy.dockerStrategy.dockerfilePath">
+                          <label for="dockerfilePath">Dockerfile Path</label>
+                          <div>
+                            <input class="form-control"
+                              id="dockerfilePath"
+                              name="dockerfilePath"
+                              type="text"
+                              ng-model="updatedBuildConfig.spec.strategy.dockerStrategy.dockerfilePath"
+                              autocorrect="off"
+                              autocapitalize="off"
+                              spellcheck="false">
+                          </div>
+                        </div>
+                      </div>
+
+                      <div class="form-group" ng-if="strategyType === 'Docker'">
+                        <div class="checkbox">
+                          <label>
+                            <input type="checkbox" ng-model="options.noCache"/>
+                            Execute docker build without reusing cached instructions.
+                            <span class="help action-inline">
+                              <a href>
+                                <i class="pficon pficon-help" data-toggle="tooltip" aria-hidden="true" data-placement="right" data-original-title="Will run the docker build with '--no-cache=true' flag">
+                                </i>
+                              </a>
+                            </span>
+                          </label>
+                        </div>
+                      </div>
+
+                      <div class="form-groups" ng-if="sources.binary && updatedBuildConfig.spec.source">
+                        <div class="form-group">
+                          <label for="binaryAsBuild">
+                            Binary Input As File
+                            <span class="help action-inline">
+                              <a href>
+                                <i class="pficon pficon-help" data-toggle="tooltip" data-placement="right" aria-hidden="true" data-original-title="Indicates that the provided binary input should be considered a single file within the build input. For example, specifying 'webapp.war' would place the provided binary as '/webapp.war' for the builder. If left empty, the Docker and Source build strategies assume this file is a zip, tar, or tar.gz file and extract it as the source. The custom strategy receives this binary as standard input. This filename may not contain slashes or be '..' or '.'."></i>
```

Comment:
can you make the part specific to the custom strategy only show up if it is a custom strategy build?   are you doing validation on this field? since we know the filename can't contain slashes or be ".."